river/preprocessing/one_hot.py

```diff
@@ -141,3 +231,65 @@ def transform_one(self, x, y=None):
                 oh[f"{i}_{xi}"] = 1
 
         return oh
+
+    # Mini-batch methods
+
+    @staticmethod
+    def _encode_1d(data, prefix, categories=None, sparse=False):
```

Comment:
Before I dive into the logic here, do you know why `pd.get_dummies` isn't sufficient here?

Reply:
Albeit I don't remember the reasoning i had exactly... 😅 

Probably it was related to the fact that `pd.get_dummies` will encode every value, regardless of what the transformer has or hasn't seen. Which would require some potentially ugly filtering and whatnot. Also it would likely be slow and memory inefficient because a lot of `pandas` abstractions are carried around in the reference implementation (`pd.get_dummies`)

So here I've kinda reverse-engineered the reference implementation, thrown away unnecessary parts and tailored the rest to our application. Wanted to keep it as close to `numpy` as reasonable (like, `pd.Categorical` seemed too powerful to abandon)